### PR TITLE
Test cleanup

### PR DESCRIPTION
**Pull request by loffek**
Define topic, partition, offset etc. once.
And initialise the ObjectMapper and Serde to test before each test.

### CODE REVIEW

File: src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java [hunk 1]
```diff
@@ -2,120 +2,110 @@
...
+    private static RecordJSONSerde serde;
+
+    @BeforeClass
+    public static void beforeAll() throws Exception {
+        // encoding here is not really important, we just want some bytes
+        keyBytes = "test-key".getBytes("UTF-8");
+        valueBytes = "test-value".getBytes("UTF-8");
+        // using Base64 as encoding in the json is part of our Record serialization format however:
+        keyBase64 = Base64.getEncoder().encodeToString(keyBytes);
+        valueBase64 = Base64.getEncoder().encodeToString(valueBytes);
```
Comment (maciejmiedzianowski) [inline src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java:38]: you can set these values directly in field definition instead of using @BeforeClass, I think 🤔

File: src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java [hunk 2]
```diff
@@ -2,120 +2,110 @@
...
+    }
+
+    @Before
```
Comment (maciejmiedzianowski) [inline src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java]: btw we can try to use JUnit 5(instead of JUnit 4) - besides some features, for example this annotation has better name in JUnit5 which is @BeforeEach - more readable

File: src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java [hunk 3]
```diff
@@ -2,120 +2,110 @@
...
+    public void beforeEach() throws Exception {
+        mapper = new ObjectMapper();
+        SimpleModule module = new SimpleModule();
+        module.addDeserializer(Record.class, new RecordJSONSerde.Deserializer());
+        module.addSerializer(Record.class, new RecordJSONSerde.Serializer());
+        mapper.registerModule(module);
+
+        serde = new RecordJSONSerde();
```
Comment (maciejmiedzianowski) [inline src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java:49]: small thing - have you ever heard about convention with naming variable `sut` (Service under test) - for me it's a good clarification what is tested in some specific class/scope

File: src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java [hunk 4]
```diff
@@ -2,120 +2,110 @@
...
-        long offset = 123;
-        // encoding here is not really important, we just want some bytes
-        byte[] key = "test-key".getBytes("UTF-8");
-        byte[] value = "test-value".getBytes("UTF-8");
-        // using Base64 as encoding here is part of our Record serialization format however:
-        String keyBase64 = Base64.getEncoder().encodeToString(key);
-        String valueBase64 = Base64.getEncoder().encodeToString(value);
         // TODO: add timestamp, timestampType, and headers
-        byte[] json = String.format("{\"topic\":\"%s\",\"partition\":%d,\"offset\":%d,\"key\":\"%s\",\"value\":\"%s\"}", topic, partition, offset, keyBase64, valueBase64).getBytes("UTF-8");
+        byte[] json = String.format("{\"topic\":\"%s\",\"partition\":%d,\"offset\":%d,\"key\":\"%s\",\"value\":\"%s\"}", topic, partition, offset, keyBase64, valueBase64).getBytes(JSON_ENCODING);
```
Comment (maciejmiedzianowski) [inline src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java:56]: maybe can we add some "test.json" file and read value from this file? what do you think?

**Review by maciejmiedzianowski — COMMENTED** [review]
Some minor comments

File: src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java (hunk 1 above)
Comment (loffek) [reply src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java:38]: Nope, they can potentially throw UnsupportedEncodingException exceptions. I could do it in a static initializer, but I would like the test to fail if it happens so...

File: src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java (hunk 2 above)
Comment (loffek) [reply src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java]: Yes. Would love that. Will try!

File: src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java (hunk 3 above)
Comment (loffek) [reply src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java:49]: Ah. That sounds nice. However, we have a bit of a weird mix here, since we are both testing the ObjectMapper and the RecordJSONSerde... 
`sutSerde` and `sutMapper`?

File: src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java (hunk 4 above)
Comment (loffek) [reply src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java:56]: I tried to add a helper, but then I realised that the json is different in almost every test, especially when testing different edge-cases like missing fields, null fields etc... But I don't like this super long line of weird formatting either... 😏

File: src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java (hunk 3 above)
Comment (maciejmiedzianowski) [reply src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java:49]: hmm, so maybe it suggests that this class should be separated into two test suits? 🤔if no and we still want to have this in single file -  I'm okay with sutSerde and sutMapper 👍

File: src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java (hunk 4 above)
Comment (maciejmiedzianowski) [reply src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java:56]: so the other idea is to extract this to separated file(ex. TestJsonFiles, TestJsonRecord etc.)with static fields like `JSON_WITH_MISSING_FIELD` and then we'll avoid such long lines.

**Review by maciejmiedzianowski — APPROVED** [review]

File: src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java (hunk 4 above)
Comment (loffek) [reply src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java:56]: Fixed this in a follow up PR

File: src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java (hunk 3 above)
Comment (loffek) [reply src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java:49]: Fixed in a follow up PR
